File: gcpc2022/kokids/submissions/accepted/Jannik_slow.java
import java.util.*;

public class Jannik_slow {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int n = in.nextInt(), k = in.nextInt();
		String s = in.next();
		int num = 0; // number of known steps
		int fallen = 0;
		for (int i = 0; i < k; i++) { // just simulate each player
			char last = 'R';
			for (int p = 0; p < n; p++) {
				if (p < num) {
					// ok
					last = s.charAt(p);
				} else {
					num++;
					if (s.charAt(p) != last) { // ok
						last = s.charAt(p);
					} else { // fall
						fallen++;
						break;
					}
				}
			}
		}
		System.out.println(k-fallen);
	}
}

File: gcpc2022/kokids/submissions/accepted/jannik_slow.cpp
#include <iostream>
#include <vector>

using namespace std;
using ll = long long;

int main() {
	ll n,k; cin >> n >> k;
	string s; cin >> s;
	ll num = 0; // number of known steps
	ll fallen = 0;
	for (ll i = 0; i < k; i++) { // just simulate each player
		char last = 'R';
		for (ll p = 0; p < n; p++) {
			if (p < num) {
				// ok
				last = s[p];
			} else {
				num++;
				if (s[p] != last) { // ok
					last = s[p];
				} else { // fall
					fallen++;
					break;
				}
			}
		}
	}
	cout << (k-fallen) << endl;
	return 0;
}
